src/NeoSharp.Core/VM/StateReader.cs

```diff
@@ -200,6 +200,11 @@ public void Dispose()
 
         protected bool CheckWitness(IExecutionEngine engine, UInt160 hash)
         {
+            if (hash.Equals(UInt160.Zero))
+            {
+                return true;
```

Comment:
Why `true`?

Reply:
The test is failing because `engine.MessageProvider` and `_transactionOperationsManager` are null. I'm not sure how to correctly perform the witness check in the unit tests, or whether to create a `InvocationTransaction` instance to check the hash against, or whether mock out the witness checking altogether. I could change it to null checks that return false, but the expected values in SmartVote.json would change a lot because the script would stop executing when `StateReader.CheckWitness` returns false

Reply:
We should fix the test, this code can't return true on `hash.Zero`. Imagine one address for burn tokens (usually zero) and with this code, you can transfer funds from this address to yours.